core/domain/summary_services_test.py

```diff
@@ -879,3 +879,132 @@ def test_requiring_public_activities_to_be_public_succeeds(self):
                 feconf.ACTIVITY_TYPE_EXPLORATION, self.EXP_ID_0),
             activity_domain.ActivityReference(
                 feconf.ACTIVITY_TYPE_COLLECTION, self.COL_ID_2)])
+
+
+class CollectionNodeMetadataDictsTest(
+        exp_services_test.ExplorationServicesUnitTests):
+    """Test functions for getting collection node metadata dicts."""
+
+    ALBERT_EMAIL = 'albert@example.com'
+    ALBERT_NAME = 'albert'
+
+    BOB_EMAIL = 'bob@example.com'
+    BOB_NAME = 'bob'
+
+    EXP_ID1 = 'eid1'
+    EXP_ID2 = 'eid2'
+    EXP_ID3 = 'eid3'
+    EXP_ID4 = 'eid4'
+    EXP_ID5 = 'eid5'
+
+    def setUp(self):
+        super(CollectionNodeMetadataDictsTest, self).setUp()
+
+        self.albert_id = self.get_user_id_from_email(self.ALBERT_EMAIL)
+        self.signup(self.ALBERT_EMAIL, self.ALBERT_NAME)
+        self.bob_id = self.get_user_id_from_email(self.BOB_EMAIL)
+        self.signup(self.BOB_EMAIL, self.BOB_NAME)
+
+        self.save_new_valid_exploration(self.EXP_ID1, self.albert_id)
+
+        exp_services.update_exploration(
```

Comment:
Instead of doing all this, why not just pass additional args to save_new_valid_exploration? Have a look at the definition of that method.